pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -609,3 +621,31 @@ func translateFault(in *networking.HTTPFaultInjection) *xdshttpfault.HTTPFault {
 
 	return &out
 }
+
+func getHashPolicy(configStore model.IstioConfigStore, hostname model.Hostname) *route.RouteAction_HashPolicy {
+	if configStore == nil {
+		return nil
+	}
+
+	destinationRule := configStore.DestinationRule(hostname)
+	if destinationRule == nil {
+		return nil
+	}
+
+	rule := destinationRule.Spec.(*networking.DestinationRule)
+	consistentHash := rule.GetTrafficPolicy().GetLoadBalancer().GetConsistentHash()
+	if consistentHash == nil {
+		return nil
+	}
+
+	cookie := consistentHash.GetHttpCookie()
```

Comment:
please convert this into a switch case and add hash policy for http header or source IP..